app/src/main/java/io/homeassistant/companion/android/widgets/graph/GraphWidgetConfigureActivity.kt

```diff
@@ -0,0 +1,308 @@
+        }
+    }
+
+    private val entityDropDownOnItemClick =
+        AdapterView.OnItemClickListener { parent, _, position, _ ->
+            selectedEntity = parent.getItemAtPosition(position) as Entity<Any>?
+            if (binding.label.text.isNullOrBlank() || labelFromEntity) {
+                selectedEntity?.friendlyName?.takeIf { it != selectedEntity?.entityId }?.let { name ->
+                    binding.label.removeTextChangedListener(labelTextChanged)
+                    binding.label.setText(name)
+                    labelFromEntity = true
+                    binding.label.addTextChangedListener(labelTextChanged)
+                }
+            }
+        }
+
+    private val labelTextChanged = object : TextWatcher {
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            // Not implemented
+        }
+
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+            // Not implemented
+        }
+
+        override fun afterTextChanged(s: Editable?) {
+            labelFromEntity = false
+        }
+    }
+
+    private fun onAddWidget() {
+        if (appWidgetId == AppWidgetManager.INVALID_APPWIDGET_ID) {
+            showAddWidgetError()
+            return
+        }
+        try {
+            val context = this@GraphWidgetConfigureActivity
+
+            // Set up a broadcast intent and pass the service call data as extras
+            val intent = Intent()
+            intent.action = BaseWidgetProvider.RECEIVE_DATA
+            intent.component = ComponentName(context, GraphWidget::class.java)
+
+            intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+
+            intent.putExtra(
+                GraphWidget.EXTRA_SERVER_ID,
+                selectedServerId!!
+            )
+
+            val entity = if (selectedEntity == null) {
+                binding.widgetTextConfigEntityId.text.toString()
+            } else {
+                selectedEntity!!.entityId
+            }
+            if (entity !in entities[selectedServerId].orEmpty().map { it.entityId }) {
+                showAddWidgetError()
+                return
+            }
+            intent.putExtra(
+                GraphWidget.EXTRA_ENTITY_ID,
+                entity
+            )
+
+            intent.putExtra(
+                GraphWidget.EXTRA_LABEL,
+                binding.label.text.toString()
+            )
+            val sliderTimeRange = binding.timeRange.value
+
+            val graphTimeRange = if (sliderTimeRange == 0F) {
+                24
+            } else {
+                sliderTimeRange.toInt()
+            }
+
+            intent.putExtra(GraphWidget.EXTRA_TIME_RANGE, graphTimeRange)
+
+            context.sendBroadcast(intent)
+
+            repository.updateWidgetLastLabel(appWidgetId, binding.label.text.toString())
+            repository.updateWidgetTimeRange(appWidgetId, binding.timeRange.value.toInt())
```

Comment:
Why save these here? They are already saved by the widget receiving the broadcast.